addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+
+import logging
+
+from collections import defaultdict
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    u_mark = fields.Boolean(
+        "Marked",
+        default=True,
+        index=True,
+        help="Pickings that are unused after refactoring are unmarked and deleted",
+    )
+
+    def get_unmarked_pickings(self):
+        """
+        Find and yield unmarked (u_mark = False) pickings.
+        Search in self when it is set, otherwise search all pickings.
+        """
+        PickingType = self.env["stock.picking.type"]
+
+        domain = [
+            ("u_mark", "=", False),
+            ("is_locked", "=", True),
+        ]
+
+        if self:
+            domain.append(("id", "in", self.ids))
+        else:
+            picking_types = PickingType.search([("u_auto_unlink_empty", "=", True)])
+            if picking_types:
+                domain.append(("picking_type_id", "in", picking_types.ids))
+
+        while True:
+            records = self.search(domain, limit=1000)
+            if not records:
+                break
+            yield records
+
+    def _get_default_new_picking_for_group_values(
+        self, group, picking_type, location, dest_location
+    ):
+        """
+        Return a dict of values for a new picking based on the supplied 
+        Procurement Group Picking Type, Location and Destination Location.
+        """
+        values = {
+            "group_id": group.id,
+            "picking_type_id": picking_type.id,
+            "location_id": location.id,
+            "location_dest_id": dest_location.id,
+        }
+
+        return values
+
+    def _remove_misleading_values(self, **kwargs):
+        """
+        If any of the fields in kwargs is set and its value is different
+        than the new one, set the field value to False to avoid misleading values.
+        E.g. if picking.origin is 'ASN001' and kwargs contains origin with value 
+        'ASN002', picking.origin is set to False.
+        """
+        self.ensure_one()
+
+        for field, value in kwargs.items():
+            if field in self:
+                current_value = self[field]
+
+                if isinstance(current_value, models.BaseModel):
+                    current_value = current_value.id
+                if current_value and current_value != value:
+                    self[field] = False
+
+    @api.model
+    def _new_picking_for_group(self, group_key, moves, **kwargs):
+        """
+        Find existing picking for the supplied group, if none found create a new one.
+        Assign the moves to the picking and return it.
+        """
+        ProcurementGroup = self.env["procurement.group"]
+
+        picking_type = moves.mapped("picking_type_id")
```

Comment:
We don't need the ```mapped()``` for relationships in odoo13

Reply:
Have now removed this.